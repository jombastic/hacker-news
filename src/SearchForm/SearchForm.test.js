import * as React from 'react';

import {
  render,
  screen,
  fireEvent,
  act,
} from '@testing-library/react';

import { SearchForm } from './SearchForm';

describe('SearchForm', () => {
  const searchFormProps = {
    searchTerm: 'React',
    onSearchInput: jest.fn(),
    onSearchSubmit: jest.fn()
  };

  it('renders the input field with its value', () => {
    render(<SearchForm {...searchFormProps} />);

    expect(screen.getByDisplayValue('React')).toBeInTheDocument();
  });

  it('renders the correct label', () => {
    render(<SearchForm {...searchFormProps} />);

    expect(screen.getByLabelText(/Search/)).toBeInTheDocument();
  });

  it('calls onSearchInput on input field change', () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.change(screen.getByDisplayValue('React'), {
      target: { value: 'Redux' },
    });

    expect(searchFormProps.onSearchInput).toHaveBeenCalledTimes(1);
  });

  it('calls onSearchSubmit on button submit click', () => {
    render(<SearchForm {...searchFormProps} />);

    fireEvent.submit(screen.getByRole('button'));

    expect(searchFormProps.onSearchSubmit).toHaveBeenCalledTimes(1);
  });

  it('renders snapshot', () => {
    const { container } = render(<SearchForm {...searchFormProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

