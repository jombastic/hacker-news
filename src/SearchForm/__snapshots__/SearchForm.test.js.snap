// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders snapshot 1`] = `
<form
  class="sc-jSUZER mefOQ"
>
  <label
    class="sc-bcXHqe cNyZAq"
    for="search"
  >
    <strong>
      Search:
    </strong>
  </label>
  Â 
  <input
    class="sc-gswNZR cRcRKt"
    id="search"
    type="text"
    value="React"
  />
  <button
    class="sc-dkrFOg sc-hLBbgP gPejQn gDcizE"
    type="submit"
  >
    Submit
  </button>
</form>
`;
